### PR TITLE
Updates to support RESTful translation service

### PR DESCRIPTION
**Pull request by hadleynet**
Some of the commits are showing up twice, I think this is a result of my rebasing the translation_service branch on master today.

### CODE REVIEW

**Comment by cmoesel** [comment]
I'll take a look.  Regarding the double commits, I expect this is because you mixed regular merges (see all the commits that start with "Merge branch") with rebase merges.  In general, you should try to avoid the regular merges (e.g., "git merge origin/master") and just pull with rebase instead (e.g., "git pull --rebase origin master").

File: Src/cql-lm/schema/elm/cqlannotations.xsd [hunk 1]
```diff
@@ -22,4 +22,61 @@
...
+  </xs:simpleType>
+  <xs:complexType name="CqlToElmError">
+    <xs:annotation>
+      <xs:documentation>Represents CQL to ELM conversion errors</xs:documentation>
+    </xs:annotation>
+    <xs:complexContent>
+      <xs:extension base="Locator">
+        <xs:attribute name="library" type="xs:string" use="required">
+          <xs:annotation>
+            <xs:documentation>The line number in the CQL where the error started</xs:documentation>
```
Comment (cmoesel) [inline Src/cql-lm/schema/elm/cqlannotations.xsd]: This description ("The line number...") doesn't seem to match the attribute name ("library").

If it's really intended to correspond to the "line number", how does this differ from the base `Locator`'s `startLine` attribute?

If it's really intended to be "library", then I'd ask if it's really needed-- as the library will already be specified in the ELM document that contains the annotation (note that `Locator` already assumes current library).  If it's not intended to be the current library, then how does it differ from `targetIncludeLibraryId`?

File: Src/cql-lm/schema/elm/cqlannotations.xsd [hunk 2]
```diff
@@ -22,4 +22,61 @@
...
+          </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="message" type="xs:string" use="required">
+          <xs:annotation>
+            <xs:documentation>A human-readable description of the error</xs:documentation>
+          </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="errorType" type="ErrorType" use="required">
+          <xs:annotation>
+            <xs:documentation>The character position in the line where the error
```
Comment (cmoesel) [inline Src/cql-lm/schema/elm/cqlannotations.xsd]: Clearly the wrong description ("The character position...")

File: Src/cql-lm/schema/elm/library.xsd [hunk 3]
```diff
@@ -1,171 +1,176 @@
 <?xml version="1.0"?>
 <!-- This file defines the root library element and type for the ELM schema. -->
```
Comment (cmoesel) [inline Src/cql-lm/schema/elm/library.xsd]: I can't tell what changed or did not change in this file (due to all the reformatting).  Did anything change?  If so, what?

Comment (hadleynet) [inline Src/cql-lm/schema/elm/library.xsd]: I don't think anything changed, just the indenting and some line wrapping on the documentation.

**Pull request merged**